arch/cortex-m4/src/lib.rs

```diff
@@ -66,30 +66,37 @@ pub unsafe extern "C" fn generic_isr() {
     mov r1, sp
     ldr r1, [r1, #4]
     stmia r1, {r4-r11}
+
+    /* Set thread mode to privileged */
+    mov r0, #0
+    msr CONTROL, r0
+
+    movw LR, #0xFFF9
+    movt LR, #0xFFFF
 _ggeneric_isr_no_stacking:
     /* Find the ISR number by looking at the low byte of the IPSR registers */
     mrs r0, IPSR
     and r0, #0xff
     /* ISRs start at 16, so substract 16 to get zero-indexed */
     sub r0, #16
 
-    /* INTERRUPT_TABLE contains function pointers, which are word sized, so
-     * multiply by 4 (the word size) */
-    lsl r0, r0, #2
 
-    ldr r1, =INTERRUPT_TABLE
-    ldr r0, [r1, r0]
+    /* r1 = NVIC.ICER[r0 / 32] */
```

Comment:
From this comment, I expect one of the instructions below to be a memory read?

Reply:
OK, it's taking an address: `r1 = &NVIC.ICER[r0 / 32]`